extensions/interactions/NumberWithUnits/directives/oppia-interactive-number-with-units.directive.ts

```diff
@@ -99,7 +99,14 @@ angular.module('oppia').directive('oppiaInteractiveNumberWithUnits', [
                   FORM_ERROR_TYPE, false);
               }
             });
-            ctrl.answer = '';
+            if ($attrs.savedSolution !== undefined) {
+              let savedSolution = JSON.parse($attrs.savedSolution);
+              savedSolution = NumberWithUnitsObjectFactory.fromDict(
+                savedSolution).toString();
+              ctrl.answer = savedSolution;
+            } else {
+              ctrl.answer = '';
+            }
```

Comment:
In other places you do not have the `else` part, why is it needed here?

Reply:
This is just the default value that the variable needs to be populated with if the solution hasn't been provided. I have used this for other interactions as well but I've been able to use ternary conditions there but since here we need to use the object factory, the ternary condition would've looked too complex which is why I added explicit if-else statements.